lockfile/src/parse_depfile.rs

```diff
@@ -80,10 +83,12 @@ pub fn parse_depfile(
     let pseudopath = Path::new(&path);
     let maybe_lockfile = parser.is_path_lockfile(pseudopath);
     let maybe_manifest = parser.is_path_manifest(pseudopath);
+    let lockfile_likely = maybe_lockfile || !maybe_manifest;
+    let manifest_likely = !maybe_lockfile || maybe_manifest;
```

Comment:
Some of our lockfile generators completely ignore the filename provided and just call the relevant tool in the directory. This can lead to weird successes... For example, in the CLI repository:

```
> ./target/debug/phylum parse -t npm Cargo.toml
Generating lockfile for manifest "Cargo.toml" using Npm…
[]                                                                                                                                                                                                                                                                                                                              
```

Reply:
I preferred when these variables were not extracted, since this does not properly capture what they represent. `maybe_lockfile` signifies that it's likely a lockfile. The `!maybe_manifest` does not make it more likely that it is a lockfile.

These variables are implying a certainty where there is none. We in fact know **less** about a file that is neither lockfile nor manifest, we're just trying everything as a hail mary.

Reply:
The variables will be put back so `lockfile_likely` and `manifest_likely` are no longer used.

@kylewillmon, are you suggesting a change is needed here? Like, maybe the lockfile generators need to be updated to _not_ ignore the filename provided?

Reply:
The cargo generator should never be run when the type is explicitly specified as npm. It should first try parsing as npm lockfile and then try npm generation.

Reply:
> It should first try parsing as npm lockfile and then try npm generation.

That is exactly what happens. The unexpected behavior here is that npm generation **succeeds**

Reply:
Oh I didn't even realize that.

Reply:
A more detailed comment was added here but no code related to the behavior was changed. If code changes are desired, I will need help to do it.

Reply:
> Like, maybe the lockfile generators need to be updated to not ignore the filename provided?

I do think this code change is warranted. Updating the generators would fix this issue.

Reply:
The lockfile generators have been updated.